woocommerce/woocommerce-ios
# Blaze: Add highlight banner on the product list

Closes: #10124 


## Description

This PR adds the Blaze banner on the product list screen following the requirements in pecCkj-MD-p2. Changes include:
- Added a user defaults key to keep track of the banner dismissal event. 
  - This key is only reset when logging out - it can be annoying to show the banner repeatedly, so I don't reset it when switching stores.
- Updated `ProductListViewModel` to check if the Blaze banner should be displayed and update its visibility when necessary.
- Updated `ProductsViewController` to show the Blaze banner in the table header when there is no loading error and at least one product. If product fetch fails, the error banner is displayed instead.
  - I'm only checking if there is any product loaded. It's not straightforward to check for published products, as it may require fetching and checking all products. Please feel free to advise any solution you see fit. 
- Updated `BlazeBanner` to add an optional divider and spacer to match the design.
- Updated `BlazeBannerHostingController` to handle actions on the banner. 
  - I'm showing only one button on the dismiss alert because the Cancel button seems redundant. Confirming with the designer in [mBhyoV7OPhFnOugxLy6SMS-fi-87:11708] - will update later if they disagree.

## Testing instructions

- Log in as an admin to a public WPCom site with at least one product to ensure you are eligible to create a Blaze campaign.
- Navigate to the Products tab, notice a banner is displayed to highlight Blaze features.
- Notice in Xcode console: `🔵 Tracked blaze_entry_point_displayed, properties: [AnyHashable("source"): "product_list",...]`
- Tap Try Blaze now, the app should navigate to the Blaze flow. An event should be tracked: `🔵 Tracked blaze_entry_point_tapped, properties: [AnyHashable("source"): "product_list",..]`
- Subsequent `blaze_flow_*` events should has the correct source `product_list`.
- Continue to complete creating a campaign (testing tips pe5sF9-1yI-p2#testing-tips) —> an event should be logged 🔵 Tracked blaze_flow_completed, properties: [AnyHashable("source"): "product_list", AnyHashable("current_step"): "step-5”, …]
- Navigate back, notice the Blaze banner is no longer on the product list, even after switching stores back and forth.
- Log out of the app and log back in to make the banner appear again.
- Tap "x" to dismiss the banner. Notice in Xcode console: `🔵 Tracked blaze_banner_dismissed, properties: [ AnyHashable("source"): "product_list",...]`
- Tap Got it on the presented alert, the Blaze banner should be dismissed and will not be available again unless after switching to a different eligible store.

## Screenshots

| Banner | When dismissed |
| ----- | ----- |
| <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/26c1e141-1dfc-454a-9d55-5f49bc0dd2ef" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/b163c63c-b9e5-4c08-bc1b-cab1ef82fcd0" width=320 /> |



---
